test/helpers/log-exploration-api.go

```diff
@@ -0,0 +1,218 @@
+}
+func (l *logExplorationAPI) HasApplicationLogs(timeToWait time.Duration) (bool, error) {
+
+	logs, err := l.getLogs()
+	if err != nil {
+		return false, err
+	}
+	if len(logs) <= 1 {
+		time.Sleep(180 * time.Second)
+		logs, err = l.getLogs()
+		if err == nil && len(logs) > 1 {
+			return true, nil
+		} else {
+			return false, err
+		}
+	}
+	return true, nil
+}
+
+func (l *logExplorationAPI) getLogs() ([]interface{}, error) {
+	options := metav1.ListOptions{
+		LabelSelector: "component=elasticsearch",
+	}
+
+	pods, err := l.tc.KubeClient.CoreV1().Pods(OpenshiftLoggingNS).List(context.TODO(), options)
+	if err != nil {
+		return nil, err
+	}
+	if len(pods.Items) == 0 {
+		return nil, errors.New("No pods found for elasticsearch")
+	}
+	clolog.V(3).Info("Pod ", "PodName", pods.Items[0].Name)
+	stdout, err := l.tc.PodExec(OpenshiftLoggingNS, pods.Items[0].Name, "elasticsearch", []string{"curl", "http://logexplorationapi:8080/logs?maxlogs=" + strconv.Itoa(maxLogsLimit)})
+	if err != nil {
+		log.Println("error while fetching logs is: ", err)
+		return nil, err
+	}
+	var jsonObjs map[string]interface{}
+
+	err = json.Unmarshal([]byte(stdout), &jsonObjs)
+	if err != nil {
+		log.Println("cannot unmarshall the response body", err)
+		return nil, err
+	}
+	objSlice, ok := jsonObjs["Logs"].([]interface{})
+	if !ok {
+		log.Println("cannot convert the JSON objects", err)
+		return nil, err
+	}
+	return objSlice, nil
+}
+func (l *logExplorationAPI) HasInfraStructureLogs(timeToWait time.Duration) (bool, error) {
+	panic("implement me")
+}
+
+func (l *logExplorationAPI) HasAuditLogs(timeToWait time.Duration) (bool, error) {
+	panic("implement me")
+}
+
+func (l *logExplorationAPI) GrepLogs(expr string, timeToWait time.Duration) (string, error) {
+	panic("implement me")
+}
+
+func (l *logExplorationAPI) RetrieveLogs() (map[string]string, error) {
+	panic("implement me")
+}
+
+func (l *logExplorationAPI) ClusterLocalEndpoint() string {
+	panic("implement me")
+}
+
+func (tc *E2ETestFramework) DeleteLogExplorationAPI(api logExplorationAPI) error {
+	var zerograce int64
+	opts := metav1.DeleteOptions{
+		GracePeriodSeconds: &zerograce,
+	}
+	return tc.KubeClient.AppsV1().Deployments(OpenshiftLoggingNS).Delete(context.TODO(), api.deployment.Name, opts)
+}
+
+func (tc *E2ETestFramework) DeployLogExplorationAPI() (deployment *apps.Deployment, err error) {
+	logExploration := &logExplorationAPI{
+		tc: tc,
+	}
+	httpGetStructLivenessProbe := &corev1.HTTPGetAction{
+		Path: "/ready",
+		Port: intstr.FromInt(8080),
+	}
+	httpGetStructReadinessProbe := &corev1.HTTPGetAction{
+		Path: "/health",
+		Port: intstr.FromInt(8080),
+	}
+	handlerStructLivenessProbe := &corev1.Handler{
+		HTTPGet: httpGetStructLivenessProbe,
+	}
+	handlerStructReadinessProbe := &corev1.Handler{
+		HTTPGet: httpGetStructReadinessProbe,
+	}
+	container := corev1.Container{
+		Name:            "logexplorationapi",
+		Image:           "quay.io/openshift-logging/log-exploration-api:latest",
```

Comment:
@igor-karpukhin where is the CI or CPaaS/midstream for this image?

Reply:
@jcantrill we don't have one atm. I'll create a task to configure CPaaS for this repo.

Reply:
@igor-karpukhin , since this PR is merging into CLO master branch. we dont need separate cpaas pipeline for it. right?